apollo-router/src/services/execution_service.rs

```diff
@@ -174,21 +179,74 @@ impl ExecutionService {
 
         let execution_span = Span::current();
 
-        let stream = stream
-            .filter_map(move |response: Response| {
-                ready(execution_span.in_scope(|| {
-                    Self::process_graphql_response(
-                        &query,
-                        operation_name.as_deref(),
-                        &variables,
-                        is_deferred,
-                        &schema,
-                        &mut nullified_paths,
-                        response,
-                    )
-                }))
-            })
-            .boxed();
+        let stream =
+            stream
+                .map(move |mut response: Response| {
+                    // Enforce JWT expiry for deferred and subscription responses
+                    if is_deferred || is_subscription {
+                        let ts_opt = claims
+                            .as_ref()
+                            .and_then(|x: &Value| {
+                                if !x.is_object() {
+                                    tracing::error!("JWT claims should be an object");
+                                    return None;
+                                }
+                                let claims = x.as_object().expect("claims should be an object");
+                                let exp = claims.get("exp")?;
+                                if !exp.is_number() {
+                                    tracing::error!("JWT 'exp' (expiry) claim should be a number");
+                                    return None;
+                                }
+                                exp.as_i64()
+                            })
+                            .and_then(|seconds_since_epoch| {
+                                let dt = chrono::DateTime::from_timestamp(seconds_since_epoch, 0);
```

Comment:
there's no need to add the chrono dependency to do this, we can do that with std:
https://github.com/apollographql/router/pull/4170/files#diff-f12f8cb5b38073d09c64f5d9739794bbd9a8882f807d809b02e44b36f0d4109eR49-R54

Reply:
done: d30a606f91ae4400f256f98f9781e44f666997d4